File: rust/sum_bytes.rs
use std::io::{Read, Write};

fn main() {
    let mut input = Vec::new();
    std::io::stdin().read_to_end(&mut input).unwrap();

    let acc: u32 = input.iter().map(|&b| b as u32).sum();

    std::io::stdout().write_all(&acc.to_le_bytes()).unwrap();
}

File: go/sum_bytes.go
package main

import (
	"encoding/binary"
	"io"
	"os"
)

func main() {
	input, err := io.ReadAll(os.Stdin)
	if err != nil {
		os.Exit(1)
	}

	var acc uint32
	for _, b := range input {
		acc += uint32(b)
	}

	var out [4]byte
	binary.LittleEndian.PutUint32(out[:], acc)
	if _, err := os.Stdout.Write(out[:]); err != nil {
		os.Exit(1)
	}
}
